pandas/core/arrays/period.py

```diff
@@ -270,12 +266,12 @@ def _check_compatible_with(self, other, setitem: bool = False):
     # Data / Attributes
 
     @cache_readonly
-    def dtype(self):
+    def dtype(self) -> PeriodDtype:
```

Comment:
as an aside, why isn't this method in the base class?

Reply:
It exists on ExtensionArray

Reply:
It's there, but raises AbstractMethodError. I was wondering why the base class doesn't return self._dtype. but I guess enforcing _dtype to exist as class attribute not so clear.